src/pages/Defi/components/OpportunityCard.tsx

```diff
@@ -96,22 +97,23 @@ export const OpportunityCard = ({
     dispatch({ type: WalletActions.SET_WALLET_MODAL, payload: true })
   }
 
-  if (!asset) return null
-
-  const getVaultName = (asset: Asset, provider: string, version?: string) => {
-    // Add Provider and Vault version if any
-    if (version) {
-      const providerExp = new RegExp('^' + provider, 'i')
-      if (!providerExp.test(version)) {
-        return `${asset.symbol} ${type?.replace('_', ' ')} (${provider} ${version})`
+  const getVaultName = useCallback(
+    (asset: Asset, provider: string, version?: string) => {
+      // Add Provider and Vault version if any
+      if (version) {
+        const providerExp = new RegExp('^' + provider, 'i')
+        if (!providerExp.test(version)) {
+          return `${asset.symbol} ${type?.replace('_', ' ')} (${provider} ${version})`
+        }
+        return `${asset.symbol} ${type?.replace('_', ' ')} (${version})`
       }
-      return `${asset.symbol} ${type?.replace('_', ' ')} (${version})`
-    }
 
-    return `${asset.symbol} ${type?.replace('_', ' ')}`
-  }
+      return `${asset.symbol} ${type?.replace('_', ' ')}`
+    },
+    [type],
+  )
 
-  const getOpportunityName = () => {
+  const getOpportunityName = useMemo(() => {
```

Comment:
Rename to `opportunityName` (this is now a memoized eefee, not a method so it shouldn't be prepended by a verb) or make it a `useCallback()` and call it in the JSX as before - happy with both options